mbed_greentea/mbed_greentea_cli.py

```diff
@@ -281,6 +281,11 @@ def main():
                     default=None,
                     help='Shuffle seed (If you want to reproduce your shuffle order please use seed provided in test summary)')
 
+    parser.add_option('', '--allocate-on-tag',
```

Comment:
I'm not convinced that `allocate-on-tag` captures what this does exactly. It seems to be a filter, using a tag, for devices under test. If this is correct, could you change the help text to match this? Could you also change the parameter to better describe it's function? something like `tag-filters` might be a better name.